src/schema/posts.ts

```diff
@@ -1218,6 +1254,46 @@ export const resolvers: IResolvers<any, Context> = {
 
       return data;
     },
+    relatedPosts: async (
+      _,
+      args: GQLPostRelationArgs,
+      ctx,
+      info,
+    ): Promise<ConnectionRelay<GQLPost>> => {
+      const post = await ctx.con
+        .getRepository(Post)
+        .findOneByOrFail({ id: args.id });
+      await ensureSourcePermissions(ctx, post.sourceId);
+
+      return queryPaginatedByDate(
```

Comment:
Should we do the permission check and the actual query in parallel to increase performance?

Reply:
I was thinking about it Promise.all would easily do it, but without the fact to actually cancel query not sure it is the best thing to do in regards to abuse. For example someone spamming invalid IDs to our service would have substantially bigger impact vs check ID first and then doing a related query.

I would leave it as is and then optimize if we see its needed, wdyt?

Reply:
Sure!